pkg/bpf/map_linux.go

```diff
@@ -890,6 +890,31 @@ func (m *Map) Delete(key MapKey) error {
 	return err
 }
 
+// DeleteButIgnoreMetricErrorCountInc deletes the map entry without incrementing the
+// metric error count for the given key, when the key does not exist.
+func (m *Map) DeleteButIgnoreMetricErrorCountInc(key MapKey, ignoreMetricErrorCount bool) error {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+
+	var err error
+	defer m.deleteCacheEntry(key, err)
+
+	if err = m.Open(); err != nil {
+		return err
+	}
+
+	_, errno := deleteElement(m.fd, key.GetKeyPtr())
+	if option.Config.MetricsConfig.BPFMapOps {
+		if !ignoreMetricErrorCount {
+			metrics.BPFMapOps.WithLabelValues(m.commonName(), metricOpDelete, metrics.Errno2Outcome(errno)).Inc()
+		}
```

Comment:
This `if` condition seems to be the only difference with the original `Delete()` method. It makes little sense to have this chunk at all, since each time you call `DeleteButIgnoreMetricErrorCountInc()` you pass `true` as second argument.

As I understand the initial idea was to add an argument to `Delete()` itself, but that would involve changing the numerous calls to the method, so it's simpler to add a new function. But you can probably base `Delete()` on the new function instead, to avoid having too much duplicated code:

``` go
func (m *Map) DeleteButIgnoreMetricErrorCountInc(key MapKey, ignoreMetricErrorCount bool) error {

	[...]

	if option.Config.MetricsConfig.BPFMapOps && !ignoreMetricErrorCount {
		metrics.BPFMapOps.WithLabelValues(m.commonName(), metricOpDelete, metrics.Errno2Outcome(errno)).Inc()
	}

	[...]

}

func (m *Map) Delete(key MapKey) error {
	return DeleteButIgnoreMetricErrorCountInc(key, false)
}
```

In that case the boolean argument makes sense, and we avoid replicating the code.

Another note: as I understand, in the case we want to skip the metric update, we only want to skip it if the error is `ENOENT`, to avoid silencing errors other than the key not being found. So you might be missing a check here.

Please make sure to address the other suggestions from older comments (e.g. on function name).

Reply:
@qmonnet I think this makes sense. FYI this looks like a "request changes" type feedback, but I see the "eye" icon from a review that selected to "Submit general feedback without explicit approval." Typically we try to avoid using this option to make it clear when changes should be addressed prior to merging, otherwise contributors do not get a clear signal whether action is required or not, and janitors do not get a clear signal about whether it is OK to merge. I've replied below requesting changes partially based on this and partially on the other feedback above.

Reply:
(If this is just Github UI mismatch or my misintepretation, feel free to disregard my comment but just trying to bring some clarification to the state of this PR and suggestions to improve the review process for both contributors and maintainers :-) )

Reply:
@joestringer No problem and agreed, I can't remember why I went for “comment” on this particular review. Likely I went too fast :/. Thanks for the notice :)